### ISSUE TITLE
Unable to add ingredients to recipe (4.0.X)

### ISSUE THREAD

**Issue by blakebjorn** [Brewtarget/brewtarget#810]
Been having this issue since 4.0 - I cannot add any ingredients to recipes, clicking the button does nothing (and nothing is logged when clicking) - the QPushButton signal connection warning happens after app startup, not when the ingredient window is launched, not sure if it is related.

I've tried removing my entire `~/.config/brewtarget` directory in case it was something to do with my database in particular, but the issue persisted on a clean install.

Ubuntu 24.04

brewtarget.log:

```
[13:13:39.578] (18waxgp5a8) INFO : Starting Brewtarget v 4.0.1  (app name "brewtarget" ) on  "Ubuntu 24.04 LTS"  [main.cpp:224]
[13:13:39.578] (18waxgp5a8) INFO : Built at Sun Aug 11 08:33:01 UTC 2024 on linux for linux with gcc compiler  [main.cpp:227]
[13:13:39.578] (18waxgp5a8) INFO : Log directory: "/home/blake/.config/brewtarget"  [main.cpp:230]
[13:13:39.578] (18waxgp5a8) INFO : Using Qt runtime v 5.15.13  (compiled against Qt v 5.15.13 )  [main.cpp:231]
[13:13:39.578] (18waxgp5a8) INFO : Configuration directory: "/home/blake/.config/brewtarget"  [main.cpp:232]
[13:13:39.578] (18waxgp5a8) INFO : Data directory: "/home/blake/.config/brewtarget"  [main.cpp:233]
[13:13:39.578] (18waxgp5a8) INFO : void {anonymous}::initResourceDir(QDir&) Application name "brewtarget"  [Application.cpp:350]
[13:13:39.578] (18waxgp5a8) INFO : void {anonymous}::initResourceDir(QDir&) Determined resource directory is "/usr/share/brewtarget"  [Application.cpp:381]
[13:13:39.578] (18waxgp5a8) INFO : Resource directory: "/usr/share/brewtarget"  [main.cpp:234]
[13:13:39.622] (18waxgp5a8) INFO : bool {anonymous}::ensureDirectoriesExist() Resource directory "/usr/share/brewtarget" exists  [Application.cpp:149]
[13:13:39.623] (18waxgp5a8) INFO : Locale: "en_US" (Decimal point: '.' / Thousands separator: ',' )  [Application.cpp:424]
[13:13:39.623] (18waxgp5a8) INFO : bool Application::initialize() Loading Database...  [Application.cpp:441]
[13:13:39.623] (18waxgp5a8) INFO : bool Database::load() Known DB drivers:  ("QSQLITE")  [database/Database.cpp:716]
[13:13:39.623] (18waxgp5a8) INFO : bool Database::impl::loadSQLite(Database&) dbFileName = " /home/blake/.config/brewtarget/database.sqlite "
dataDbFileName=" /usr/bin/../share/brewtarget/default_db.sqlite "  [database/Database.cpp:254]
[13:13:39.623] (18waxgp5a8) INFO : bool Database::impl::loadSQLite(Database&) SQLite version QVariant(QString, "3.45.1")  [database/Database.cpp:303]
[13:13:39.624] (18waxgp5a8) INFO : bool Database::impl::updateSchema(Database&, bool*) Schema version in DB: 12 , current schema version in code: 12  [database/Database.cpp:400]
[13:13:39.624] (18waxgp5a8) INFO : DefaultContentLoader::UpdateResult DefaultContentLoader::updateContentIfNecessary(QSqlDatabase&, QTextStream&) availableContentVersion: 4 , defaultContentAlreadyLoaded: 4  [database/DefaultContentLoader.cpp:97]
[13:13:39.637] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 125 objects from DB table recipe  [database/ObjectStore.cpp:1667]
[13:13:39.637] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 5 objects from DB table brewnote  [database/ObjectStore.cpp:1667]
[13:13:39.652] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 351 objects from DB table style  [database/ObjectStore.cpp:1667]
[13:13:39.660] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 147 objects from DB table equipment  [database/ObjectStore.cpp:1667]
[13:13:39.683] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 844 objects from DB table fermentable  [database/ObjectStore.cpp:1667]
[13:13:39.697] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 548 objects from DB table hop  [database/ObjectStore.cpp:1667]
[13:13:39.699] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 97 objects from DB table misc  [database/ObjectStore.cpp:1667]
[13:13:39.718] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 808 objects from DB table yeast  [database/ObjectStore.cpp:1667]
[13:13:39.724] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 14 objects from DB table water  [database/ObjectStore.cpp:1667]
[13:13:39.736] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 96 objects from DB table boil  [database/ObjectStore.cpp:1667]
[13:13:39.738] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 240 objects from DB table boil_step  [database/ObjectStore.cpp:1667]
[13:13:39.738] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 95 objects from DB table fermentation  [database/ObjectStore.cpp:1667]
[13:13:39.739] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 104 objects from DB table fermentation_step  [database/ObjectStore.cpp:1667]
[13:13:39.740] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 129 objects from DB table instruction  [database/ObjectStore.cpp:1667]
[13:13:39.741] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 501 objects from DB table fermentable_in_inventory  [database/ObjectStore.cpp:1667]
[13:13:39.742] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 348 objects from DB table hop_in_inventory  [database/ObjectStore.cpp:1667]
[13:13:39.742] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 97 objects from DB table misc_in_inventory  [database/ObjectStore.cpp:1667]
[13:13:39.742] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 0 objects from DB table salt_in_inventory  [database/ObjectStore.cpp:1667]
[13:13:39.744] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 727 objects from DB table yeast_in_inventory  [database/ObjectStore.cpp:1667]
[13:13:39.745] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 211 objects from DB table mash  [database/ObjectStore.cpp:1667]
[13:13:39.748] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 253 objects from DB table mash_step  [database/ObjectStore.cpp:1667]
[13:13:39.752] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 501 objects from DB table fermentable_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.754] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 273 objects from DB table hop_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.754] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 0 objects from DB table misc_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.756] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 122 objects from DB table yeast_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.756] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 0 objects from DB table salt_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.756] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 0 objects from DB table water_in_recipe  [database/ObjectStore.cpp:1667]
[13:13:39.756] (18waxgp5a8) INFO : void ObjectStore::loadAll(Database*) Read 0 objects from DB table salt  [database/ObjectStore.cpp:1667]
[13:13:40.267] (18waxgp5a8) WARNING : QObject::connect(QPushButton, Unknown): invalid nullptr parameter  [:0]
[13:13:40.267] (18waxgp5a8) WARNING : QObject::connect(QPushButton, Unknown): invalid nullptr parameter  [:0]
```

**Comment by matty0ung** [comment]
Sorry about this.  The good news is I can reproduce it locally, so I'll have a look at what's going on and work on a fix.

**Comment by matty0ung** [comment]
Just to give you an update, I have most of the fix coded locally.  The right things now happen with ingredients getting added to recipes etc, but I just need to finish off connecting up the signals and slots so that the display updates.  Hope to have a patch in the coming week.

**Comment by matty0ung** [comment]
Should be fixed in [4.0.2](https://github.com/Brewtarget/brewtarget/releases/tag/v4.0.2).
